test/gurka/test_search_filter.cc

```diff
@@ -182,3 +185,840 @@ TEST_F(SearchFilter, ExcludeRamp) {
+  std::string costing_speed_type =
+      (boost::format("/costing_options/%s/speed_types/0") % costing).str();
+
+  {
+    const std::string& req_with_heading =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"heading":0},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("M").lat()) %
+         std::to_string(closure_map.nodes.at("M").lng()) %
+         std::to_string(closure_map.nodes.at("B").lat()) %
+         std::to_string(closure_map.nodes.at("B").lng()) % costing % costing % date_type)
+            .str();
+    auto result = gurka::route(closure_map, req_with_heading, reader);
+    gurka::assert::osrm::expect_steps(result, {"KM", "HI", "FI", "EF", "BE"});
+    gurka::assert::raw::expect_path(result, {"KM", "HK", "HI", "FI", "EF", "BE"});
+  }
+
+  // Close consecutive edges CD & DE
+  {
+    LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+      close_single_edge(reader, tile, index, current, "HK", closure_map);
+      close_single_edge(reader, tile, index, current, "KM", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    const std::string& req_with_heading =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"heading":0},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("M").lat()) %
+         std::to_string(closure_map.nodes.at("M").lng()) %
+         std::to_string(closure_map.nodes.at("B").lat()) %
+         std::to_string(closure_map.nodes.at("B").lng()) % costing % costing % date_type)
+            .str();
+    EXPECT_THROW(gurka::route(closure_map, req_with_heading, reader), valhalla_exception_t);
+
+    const std::string& req_disable_exclude_closures =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"heading":0,"search_filter":{"exclude_closures":false}},{"lat":%s,"lon":%s,"search_filter":{"exclude_closures":false}}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("M").lat()) %
+         std::to_string(closure_map.nodes.at("M").lng()) %
+         std::to_string(closure_map.nodes.at("B").lat()) %
+         std::to_string(closure_map.nodes.at("B").lng()) % costing % costing % date_type)
+            .str();
+    auto result = gurka::route(closure_map, req_disable_exclude_closures, reader);
+    gurka::assert::osrm::expect_steps(result, {"HI", "FI", "EF", "BE"});
```

Comment:
This test too fails with a no-route (both with `exclude_closure:true/false` at the start location). I've tried `minimum_reachability` of 0 & 1 but that didn't seem to affect anything.

Reply:
ok yeah this is because the reach check doesnt do what we think it should do. the reach check has two phases, conservative fast mode, and hardcore full checking mode which it falls back to when the conservative estimate is a lower reach than we'd like.

i cant explain exclude_closure:true, but when its false, what happens is that conserative estimate keeps excluding closures whiel its doing the reach check and thinks that eveyrthing is reachable. to fix this we will need to add another check to the conservative check which tests whether the edge is currently closed. i can make this change and do a benchmark to see what that does to performance. the sad thing is it allready checks it in the filter but there is no clean way to find out about it (other than making the filter funciton take more options)